src/Storages/S3Queue/StorageS3Queue.cpp

```diff
@@ -112,7 +112,6 @@ StorageS3Queue::StorageS3Queue(
     , s3queue_settings(std::move(s3queue_settings_))
     , zk_path(chooseZooKeeperPath(table_id_, context_->getSettingsRef(), *s3queue_settings))
     , after_processing(s3queue_settings->after_processing)
-    , files_metadata(S3QueueMetadataFactory::instance().getOrCreate(zk_path, *s3queue_settings))
```

Comment:
reason: if an exception in constructor happened, we did not decrement the ref count.